cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -864,9 +866,15 @@ private void publishAudit(StreamId stream, AuditType auditType) {
     }
   }
 
-  private void verifyOwner(StreamId streamId, @Nullable String specifiedOwnerPrincipal) throws IOException {
-    boolean equals = Objects.equals(specifiedOwnerPrincipal, ownerAdmin.getOwnerPrincipal(streamId));
-    Preconditions.checkArgument(equals,
-                                String.format("Updating %s is not supported.", Constants.Security.PRINCIPAL));
+  private void verifyOwner(StreamId streamId,
+                           @Nullable String specifiedOwnerPrincipal) throws IOException, ConflictException {
+    // verify  owner with the effective owner principal
+    if (!Objects.equals(specifiedOwnerPrincipal, ownerAdmin.getImpersonationPrincipal(streamId))) {
+      // Not giving existing owner information as it might be unacceptable under some security scenarios
+      throw new ConflictException(String.format("The specified %s : '%s' is not the same as existing one. " +
```

Comment:
- there should not be a s]pace before the :
- it should be "not the same as the existing"
- the syntax is weird. In the original PR you had one message that looked good. I asked you to make the other ones similar to that. Instead you changed the good one to look like the other ones. 

Perhaps @awholegunch should review the messages instead of me.